src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -214,129 +429,190 @@ performFetchWithCompletionHandler:(void (^)(UIBackgroundFetchResult result))comp
+  </li>
+  <li class="nav-item">
+    <a class="nav-link" id="initial-route-swift" href="#initial-route-swift-tab" role="tab" aria-controls="initial-route-swift" aria-selected="false">Swift</a>
+  </li>
+</ul>
 
-## Hot restart/reload and debugging Dart code
+<div class="tab-content"> {% comment %} Initial route language tab start {% endcomment -%}
 
-Connect a device or launch a Simulator. Then make Flutter CLI tooling listen
-for your app to come up:
+<div class="tab-pane active" id="initial-route-objc-tab" role="tabpanel" aria-labelledby="initial-route-objc-tab" markdown="1">
 
-```bash
-$ cd some/path/my_flutter
-$ flutter attach
-Waiting for a connection from Flutter on iPhone X...
+<?code-excerpt "Creating engine" title?>
+```objectivec
+FlutterEngine *flutterEngine = [[FlutterEngine alloc] initWithName:@"my flutter engine"];
+[[flutterEngine navigationChannel] invokeMethod:@"setInitialRoute" arguments:@"/onboarding"];
+[flutterEngine run];
+@end
 ```
 
-Launch `MyApp` in debug mode from Xcode. Navigate to an area of the app that uses
-Flutter. Then turn back to the terminal, and you should see output similar to the
-following:
+</div>
 
-```text
-Done.
-Syncing files to device iPhone X...                          4.7s
+<div class="tab-pane" id="initial-route-swift-tab" role="tabpanel" aria-labelledby="initial-route-swift-tab" markdown="1">
 
-🔥  To hot reload changes while running, press "r". To hot restart (and rebuild state), press "R".
-An Observatory debugger and profiler on iPhone X is available at: http://127.0.0.1:54741/
-For a more detailed help message, press "h". To quit, press "q".
+<?code-excerpt "Creating engine" title?>
+```swift
+let flutterEngine = FlutterEngine(name: "my flutter engine")
+flutterEngine.navigationChannel.invokeMethod("setInitialRoute", arguments:"/onboarding")
+flutterEngine.run()
 ```
 
-You can now edit the Dart code in `my_flutter`, and the changes can be hot
-reloaded by pressing `r` in the terminal. You can also paste the URL above into
-your browser to use the Dart Observatory for setting breakpoints, analyzing
-memory retention and other debugging tasks.
+</div>
+
+</div>{% comment %} Initial route language tab end {% endcomment -%}
+
+will set your `dart:ui`'s [`window.defaultRouteName`]({{site.api}}/flutter/dart-ui/Window/defaultRouteName.html)
+to `"/onboarding"` instead of `"/"`.
+
+{{site.alert.note}}
+`"setInitialRoute"` on the `navigationChannel` must be called before running your
+`FlutterEngine` in order for Flutter's very first frame to use the desired
+route.
+{{site.alert.end}}
+
+{{site.alert.tip}}
+In order to change your Flutter route after the `FlutterEngine` is already
+running, use [pushRoute]({{site.api}}/objcdoc/Classes/FlutterViewController.html#/c:objc(cs)FlutterViewController(im)pushRoute:)
+or [popRoute]({{site.api}}/objcdoc/Classes/FlutterViewController.html#/c:objc(cs)FlutterViewController(im)popRoute)
+on the `FlutterViewController`.
```

Comment:
This is markedly different from setting an initial route, since it mutates the route stack, whereas `setInitialRoute` initializes the route stack to a specific spot.

Reply:
I wasn't suggesting that they are equivalent. It's in a tangential tip box.

Reply:
Added a few more words to clarify difference.